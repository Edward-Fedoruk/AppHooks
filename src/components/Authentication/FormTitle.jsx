import React from 'react'
import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'
import PropTypes from 'prop-types'

const styles = () => ({
  formTitleWrap: {
    position: "relative",
    marginBottom: "15px"
  },

  line: {
    width: "100%",
    height: "1px",
    backgroundColor: "#DADADA"
  },

  formTitle: {
    textTransform: "uppercase",
    color: "#8E8F93",
    position: "absolute",
    fontSize: "14px",
    fontWeight: "500",
    backgroundColor: "#fff",
    padding: "0 20px",
    top: "50%",
    right: "50%",
    transform: "translate(50%, -50%)",
    textAlign: "center",
    width: "max-content"
  },
})

const FormTitle = ({ classes, text }) => (
  <div className={classes.formTitleWrap}>
    <div className={classes.line}></div>
    <Typography className={classes.formTitle}>{ text }</Typography>
  </div>
)

FormTitle.propTypes = {
  text: PropTypes.string.isRequired
}

export default withStyles(styles)(FormTitle)
