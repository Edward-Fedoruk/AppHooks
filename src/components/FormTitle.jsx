import React, { Fragment } from "react"
import PropTypes from "prop-types"
import Typography from "@material-ui/core/Typography"
import { withStyles } from "@material-ui/core/styles"

const styles = () => ({
  title: {
    fontWeight: "500",
    fontSize: "25px",
    color: "#5A5B5F",
    marginBottom: "25px",
  },

  p: {
    fontSize: "16px",
    color: "#7C7D81",
  },
})

const FormTitle = ({
  classes, titleStyle, pStyle, title, paragraph,
}) => (
  <Fragment>
    <Typography
      className={classes.title}
      variant="h3"
      align="left"
      style={titleStyle}
    >
      {title}
    </Typography>

    <Typography
      className={classes.p}
      align="left"
      style={pStyle}
    >
      { paragraph }
    </Typography>
  </Fragment>
)

FormTitle.defaultProps = {
  titleStyle: {},
  pStyle: {},
}

FormTitle.propTypes = {
  classes: PropTypes.object.isRequired,
  titleStyle: PropTypes.object,
  pStyle: PropTypes.object,
  title: PropTypes.string.isRequired,
  paragraph: PropTypes.string.isRequired,
}

export default withStyles(styles)(FormTitle)
