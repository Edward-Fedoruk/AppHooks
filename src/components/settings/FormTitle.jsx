import React from "react"
import PropTypes from "prop-types"
import Typography from "@material-ui/core/Typography"
import { withStyles } from "@material-ui/core"
import IconButton from "@material-ui/core/IconButton"
import Create from "@material-ui/icons/CreateOutlined"

const styles = ({ breakpoints }) => ({
  header: {
    fontSize: "18px",
    fontWeight: "500",
    marginRight: "5px",

    [breakpoints.down(425)]: {
      paddingLeft: "24px",
    },
  },

  headerWrap: {
    display: "flex",
    alignItems: "center",
  },
})

const FormTitle = ({ classes, children, toggleForm }) => (
  <div className={classes.headerWrap}>
    <Typography className={classes.header} variant="h2" color="primary">
      { children }
    </Typography>
    <IconButton onClick={toggleForm}>
      <Create />
    </IconButton>
  </div>
)


FormTitle.propTypes = {
  classes: PropTypes.object.isRequired,
  children: PropTypes.string.isRequired,
  toggleForm: PropTypes.func.isRequired,
}

export default withStyles(styles)(FormTitle)
