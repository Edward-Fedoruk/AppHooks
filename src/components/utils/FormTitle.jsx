import React, { Fragment } from "react"
import PropTypes from "prop-types"
import Typography from "@material-ui/core/Typography"
import { withStyles } from "@material-ui/core/styles"

const styles = ({ palette }) => ({
  title: {
    fontWeight: "500",
    fontSize: "25px",
    color: palette.primary.main,
    marginBottom: "25px",
  },

  p: {
    fontSize: "16px",
    color: palette.primary.main,
  },
})

const FormTitle = ({
  classes, titleStyle, pStyle, title, paragraph,
}) => (
  <Fragment>
    <Typography
      className={classes.title}
      variant="h3"
      align="left"
      style={titleStyle}
    >
      {title}
    </Typography>

    <Typography
      className={classes.p}
      align="left"
      style={pStyle}
    >
      { paragraph }
    </Typography>
  </Fragment>
)

FormTitle.defaultProps = {
  titleStyle: {},
  pStyle: {},
}

FormTitle.propTypes = {
  classes: PropTypes.object.isRequired,
  titleStyle: PropTypes.object,
  pStyle: PropTypes.object,
  title: PropTypes.string.isRequired,
  paragraph: PropTypes.string.isRequired,
}

export default withStyles(styles)(FormTitle)
