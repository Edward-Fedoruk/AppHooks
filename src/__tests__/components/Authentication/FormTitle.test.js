import React from 'react'
import FormTitle from '../../../components/Authentication/FormTitle'
import { shallow } from "enzyme"

describe('FormTitle', () => {
  it('should be defined', () => {
    expect(FormTitle).toBeDefined()
  })
  it('should render correctly', () => {
    const tree = shallow(
      <FormTitle text={"test"} />
    )
    expect(tree).toMatchSnapshot()
  })
  it('should have text props', () => {
    const tree = shallow(
      <FormTitle text={"test"} />
    )
    expect(tree.props().text).toEqual("test")
  })
})