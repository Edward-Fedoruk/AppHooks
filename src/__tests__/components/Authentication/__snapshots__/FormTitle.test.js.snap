// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTitle should render correctly 1`] = `
<FormTitle
  classes={
    Object {
      "formTitle": "FormTitle-formTitle-2",
      "formTitleWrap": "FormTitle-formTitleWrap-1",
    }
  }
  text="test"
/>
`;
